Model/Api/DiscountCodeValidation.php

```diff
@@ -1024,4 +1014,32 @@ protected function getParentQuoteDiscountResult($couponCode, $coupon, $parentQuo
             'discount_type'   => $this->convertToBoltDiscountType($rule->getSimpleAction()),
         ];
     }
+    
+    /**
+     * @param object $address
+     *
+     * @return array
+     */
+    private function prepareAddressData($address)
+    {
+        $address = $this->cartHelper->handleSpecialAddressCases($address);
+        $region = $this->regionModel->loadByName(@$address->region, @$address->country_code);
+        $addressData = [
+                    'firstname'    => @$address->first_name,
```

Comment:
@matt-thomason Do I understand correctly we want to stop use "@" ? Do we have any doc about this?